.github/workflows/check-docs.yml

```diff
@@ -0,0 +1,38 @@
+---
+
+name: Check Docs
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+  lint-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout dsp-tools repo
+        uses: actions/checkout@v4
+      - name: Install uv, just, and dependencies
+        uses: ./.github/actions/setup
+      - name: Linting with darglint
+        run: just darglint
+      - name: check external links in docs
+        run: |
+          npm install -g markdown-link-validator
+          just check-links
+      - name: markdownlint
+        run: just markdownlint
+
+  build-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout dsp-tools repo
+        uses: actions/checkout@v4
+      - name: Install uv, just, and dependencies
+        uses: ./.github/actions/setup
```

Comment:
From an economical and ecological stand point, it doesn't make sense to spin up an entire VM, and install everything, just for building the docs. I would argue that this step can also be done inside the first job.

```suggestion
```